### ISSUE TITLE
Monospace formatting does not show some characters correctly

### ISSUE THREAD

**Issue by hyperion-cs** [telegramdesktop/tdesktop#25204]
### Steps to reproduce

1. Write a message for anyone that contains, for example, the following:
    ```rs
    let x = true || false;
    let y = if x { 5 } else { 10 };
    ```
2. Apply monospace formatting to the text above.

### Expected behaviour

Expect to see the same characters, but in monospace format.

### Actual behaviour

In actuality, we get incorrect characters in some places (this seems to relate to some NOT alphabetic characters):
![image](https://user-images.githubusercontent.com/14217598/196158694-c9ecd639-9c1e-452f-9bc5-5f702f4f3862.png)

Notably, if the text is copied, it is completely correct. Thus, this is definitely a display problem in the font.

### Operating system

Windows 10 / 21H2 / Build 19044.2130

### Version of Telegram Desktop

4.2.4 x64

### Installation source

Static binary from official website

**Comment by hyperion-cs** [comment]
@ilya-fedin, I don't think so. Look at the example of working with this font in other utilities:
![image](https://user-images.githubusercontent.com/14217598/196163589-3dea2381-d9ff-4c82-8d85-2a7daf8048d4.png)
![image](https://user-images.githubusercontent.com/14217598/196162955-692546c9-5e4f-4782-8079-08bb2704193e.png)

I checked all the font options (Light, Regular, Bold, etc.).

**Comment by ilya-fedin** [comment]
Apparently Qt reads it incorrectly on your system for whatever reason, maybe you have Arabic localization installed or something else

**Comment by hyperion-cs** [comment]
@ilya-fedin,
1. I have only Russian and English installed in the system;
2. Region/local is set to "Russian", no Arabic localization;
3. Before, everything worked correctly on the same system;
4. In the web version of Telegram text in monospace format looks correct (however, it is obvious that the web version does not use Qt).

**Comment by hyperion-cs** [comment]
@ilya-fedin, I was able to fix this problem. For those who will face it:
"_Beta: Use Unicode (UTF-8) to support language worldwide_" should be checked in the regional standards settings:
![image](https://user-images.githubusercontent.com/14217598/196167225-43864fdb-5734-4ef4-84be-15abd03e0a1a.png)

**Comment by ilya-fedin** [comment]
> * Before, everything worked correctly on the same system;

Cascadia Mono is preferred since 4.2.1, earlier Consolas was preferred. Other than adding it to the top of a vector, there were no change.

**Comment by ilya-fedin** [comment]
> 4\. In the web version of Telegram text in monospace format looks correct (however, it is obvious that the web version does not use Qt).

I'm not sure it uses Cascadia Mono either

**Comment by ilya-fedin** [comment]
> * If localization was the problem, why does the regular font (not monospace) in Telegram Desktop look correct?

Qt ¯\_(ツ)_/¯
I remember there was a report that with some update some users got weird characters instead of the timestamps. I don't know whether it's still the case, might be so.

**Comment by hyperion-cs** [comment]
@ilya-fedin,
Fortunately, I had screenshots of Telegram Desktop earlier (just before version 4.2.1). There, indeed, the Consolas font was used in monospace formatting and all the characters looked correct.
So, the problem seems to have appeared after the default Cascadia Mono font started to be used. The good thing is that it is solved by adjusting some settings in Windows.

----

>I'm not sure it uses Cascadia Mono either

Preference is given to it:
```css
--font-family-monospace: "Cascadia Mono", "Roboto Mono", "Droid Sans Mono", "SF Mono", "Menlo", "Ubuntu Mono", "Consolas", monospace;
```
And it is used:
![image](https://user-images.githubusercontent.com/14217598/196170862-ab4e32d3-82ff-4407-96ee-2b3d67e957bb.png)

**Comment by aggieben** [comment]
I seem to be having a similar issue on Windows 11; this is new - monospace has always worked just fine for me on tdesktop.  Example:
Before formatting:
![image](https://user-images.githubusercontent.com/1321/199267399-5d22d886-92d0-47d4-9a79-ccc71f95a7e0.png)

After formatting (Using Ctrl-Shift-M):
![image](https://user-images.githubusercontent.com/1321/199267500-7c32f467-b991-474e-a869-2d71365284b4.png)

It may be ok for some to use the UTF-8 locale setting, but that's a beta feature in Windows, and I've tried it before and experienced instability in various ways (don't remember details).  This feature in tdesktop ought to work without requiring users to resort to beta/experimental features in the OS.
